src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -328,14 +356,24 @@ def add_description_multiple(self, descriptions: list[str]) -> VideoSegmentResou
         self.descriptions.extend(descriptions)
         return self
 
-    def add_keyword(self, keywords: str) -> VideoSegmentResource:
-        self.keywords.append(keywords)
+    def add_description_optional(self, description: Any) -> VideoSegmentResource:
+        if not pd.isna(description):
+            self.descriptions.append(description)
+        return self
+
+    def add_keyword(self, keyword: str) -> VideoSegmentResource:
```

Comment:
@Notheturtle I noticed that the variable was in the plural. I fixed this, but it may break a script of yours.